src/context/server_events.rs

```diff
@@ -0,0 +1,217 @@
+use crate::context::Context;
+use crate::raw;
+use crate::RedisError;
+
+#[derive(Clone)]
+pub enum ServerRole {
+    Primary,
+    Replica,
+}
+
+#[derive(Clone)]
+pub enum LoadingSubevent {
+    RdbStarted,
+    AofStarted,
+    ReplStarted,
+    Ended,
+    Failed,
+}
+
+#[derive(Clone)]
+pub enum FlushSubevent {
+    Started,
+    Ended,
+}
+
+#[derive(Clone)]
+pub struct RoleChangedEventData {
+    pub role: ServerRole,
+}
+
+#[derive(Clone)]
+pub enum ServerEventData {
```

Comment:
There are several more events (even with oss), for example, `RedisModuleEvent_ClientChange`, `RedisModuleEvent_Shutdown`, etc.
But we only support a subset now?

Reply:
I will add it gradually when needed.